packages/core/include/soss/SystemHandle.hpp

```diff
@@ -72,17 +77,20 @@ class SystemHandle
   ///
   /// \param[in] types
   ///   The set of types (messages and services) that this middleware needs to
-  ///   support. The system handle should load any additional plugin libraries
-  ///   needed by this
+  ///   support. The system handle must register this types to the TypeRegistry.
   ///
   /// \param[in] configuration
   ///   The configuration specific to this system handle, as described in the
   ///   user-provided yaml input file.
   ///
+  /// \param[in] type_register
```

Comment:
```suggestion
  /// \param[in] type_registry
```
Although maybe type_registry a more specific name for this parameter can add more value